.github/workflows/tests.yml

```diff
@@ -49,7 +54,7 @@ jobs:
         with:
           enable-cache: true
       - run: uv python pin ${{ matrix.python-version }}
-      - run: uv sync --python-preference=only-managed
+      - run: uv sync --python-preference=only-managed --extra qdrant
```

Comment:
So people running just `uv sync` locally and running tests will get fails on the Qdrant tests.

We need to either:
- Add the `qdrant` extra to our `dev` dependencies in the repo root's `pyproject.toml`'s `dependency-groups`' `dev`
- `pytest.skip` the Qdrant test cases if `qdrant-client` isn't installed

Can you pick one? I vote the second bullet

Reply:
I found 1 more appropriate. Since we've parameterised tests to use both Numpy and Qdrant, `pytest.skip` would skip both if `qdrant-client` isn't installed.

The tests now use an in-memory instance. There's no need for Docker to be running.

Reply:
The in-memory mode is a Python implementation that mimics all the APIs of the Rust based Qdrant server.
It's used for prototyping and CIs.